utils.go

```diff
@@ -49,15 +51,61 @@ func ConfPath(cluster string) string {
 }
 
 // Get the keys repo to use keys.
-// Callers are responsible for removing the repo when done.
-func getKeysrepo() (string, error) {
+// url: Use specified git URL to git clone repository.
+//      If "url" is empty, then use machine's default key repo.
+// Note: Callers are responsible for removing the repo when done.
+//
+func getGitKeysrepo(url string) (string, error) {
 	dir, err := os.MkdirTemp("", "keys")
 	if err != nil {
 		return "", err
 	}
-	_, err = git.PlainClone(dir, false, &git.CloneOptions{URL: "https://github.com/project-machine/keys.git",})
+	if url != "" {
+		_, err = git.PlainClone(dir, false, &git.CloneOptions{URL: url,})
+	}
 	if err != nil {
+		os.RemoveAll(dir)   // cleanup
 		return "", err
+	} else {
+		return dir, nil
+	}
+}
+
+// Get location of the keys repo.
+// If "keysdir" is not specified, then check the env var.
+// If neither is set, then use machine's keys git repo.
+//
+func getKeysrepoLocation(keysdir string) string {
+	var location string
+
+	if keysdir != "" {
+		location = keysdir
+	} else {
+		location = os.Getenv("KEYSREPO")
+		if location == "" {
+			location = "https://github.com/project-machine/keys.git"
+		}
+	}
+
+	return location
+}
+
+func getKeysrepo(keysdir string) (string, error) {
+	// get location of keysdir to use
+	keyrepo := getKeysrepoLocation(keysdir)
+
+	// Check if a path on local disk
+	if keyrepo != "" && strings.HasPrefix(keyrepo, "/") {
+		if  PathExists(keyrepo) {
+			return keyrepo, nil
+		} else {
+			return "", fmt.Errorf("Specified keys repo %s does not exist.", keyrepo)
+		}
+	}
+
+	// Check if a github URL
+	if keyrepo != ""  && strings.HasPrefix(keyrepo, "https://github.com/") {
```

Comment:
Probably shouldn't require github, right?

Reply:
ughhh i coded this on my other laptop... something up with my vimrc... let me check